src/eth-signer/src/ic_sign.rs

```diff
@@ -62,21 +62,44 @@ pub struct IcSigner;
 
 impl IcSigner {
     /// Signs the transaction using `ManagementCanister::sign_with_ecdsa()` call.
+    /// The `tx.from` expected to be set to the canister address.
     pub async fn sign_transaction(
         &self,
         tx: &TypedTransaction,
         key_id: SigningKeyId,
         derivation_path: DerivationPath,
     ) -> Result<Signature, IcSignerError> {
         let hash = tx.sighash();
-        let hash_bytes = hash.as_fixed_bytes();
+        let digest = hash.as_fixed_bytes();
+        let tx_from = tx.from().ok_or(IcSignerError::FromAddressNotPresent)?;
+        let mut signature = Self
+            .sign_digest(tx_from, *digest, key_id, derivation_path)
+            .await?;
+
+        // For non-legacy transactions recovery id should be updated.
+        // Details: https://eips.ethereum.org/EIPS/eip-155.
+        signature.v += match &tx {
+            TypedTransaction::Legacy(_) => 27,
+            _ => tx.chain_id().unwrap_or_default().as_u64() * 2 + 35,
```

Comment:
` If you do, then the v of the signature MUST be set to {0,1} + CHAIN_ID * 2 + 35 where {0,1} is the parity of the y value of the curve point for which r is the x-value in the secp256k1 signing process`
Do we check this parity?

Reply:
Also, is this a todo for non legacy transaction?

Reply:
For legacy transactions, it's `{0,1} + 27`.
For legacy transactions, it's `{0,1} + CHAIN_ID * 2 + 35`.

This parity check is done inside the `sign_digest` method.